src/coreclr/jit/lowerxarch.cpp

```diff
@@ -5184,28 +4944,52 @@ GenTree* Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case TYP_LONG:
+        case TYP_ULONG:
         case TYP_DOUBLE:
         {
+            GenTree* op2 = node->Op(2);
+
+            if (varTypeIsLong(simdBaseType) && comp->compOpportunisticallyDependsOn(InstructionSet_SSE41_X64))
+            {
+                // We will be constructing the following parts:
+                //   ...
+                //   idx  =    CNS_INT       int    1
+                //          /--*  tmp1 simd16
+                //          +--*  op2  T
+                //          +--*  idx  int
+                //   node = *  HWINTRINSIC   simd16 T Insert
+
+                // This is roughly the following managed code:
+                //   ...
+                //   return Sse41.X64.Insert(tmp1, op2, 0x01);
+
+                idx = comp->gtNewIconNode(0x01, TYP_INT);
+                BlockRange().InsertBefore(node, idx);
+
+                node->ResetHWIntrinsicId(NI_SSE41_X64_Insert, comp, tmp1, op2, idx);
+                break;
+            }
+
             // We will be constructing the following parts:
             //   ...
             //          /--*  op2  T
             //   tmp2 = *  HWINTRINSIC   simd16 T CreateScalarUnsafe
             //          /--*  tmp1 simd16
             //          +--*  tmp2 simd16
-            //   node = *  HWINTRINSIC   simd16 T MoveLowToHigh
+            //   node = *  HWINTRINSIC   simd16 T UnpackLow
 
             // This is roughly the following managed code:
             //   ...
             //   var tmp2 = Vector128.CreateScalarUnsafe(op2);
-            //   return Sse.MoveLowToHigh(tmp1, tmp2);
+            //   return Sse.UnpackLow(tmp1, tmp2);
 
             assert(comp->compIsaSupportedDebugOnly(InstructionSet_SSE2));
 
             tmp2 = InsertNewSimdCreateScalarUnsafeNode(TYP_SIMD16, op2, simdBaseJitType, 16);
             LowerNode(tmp2);
 
-            node->ResetHWIntrinsicId(NI_SSE_MoveLowToHigh, tmp1, tmp2);
```

Comment:
Changing this to `UnpackLow` shows up as a regression in a few places, because it's one byte larger, but it enables other optimizations since `unpcklpd` takes a memory operand and `movlhps` doesn't:

```diff
-       movups   xmm0, xmmword ptr [reloc @RWD00]
-       movlhps  xmm1, xmm0
+       unpcklpd xmm1, xmmword ptr [reloc @RWD00]
```

Reply:
This should probably be peepholed back to `vmovlhps` if both are from register.

Reply:
I was thinking the same but would rather save that for a followup.  llvm has a replacement list of equivalent instructions that have different sizes, and `unpcklpd` is on it, as are things like `vpermilps`, which is replaced by `pshufd`.

It's worth having a discussion about whether we'd also want to do replacements that switch between float and integer domains. I'll open an issue.

Reply:
I looked at this again, and there's actually only a size difference for legacy SSE encoding, so it's probably not worth special casing.